controllers/release/adapter.go

```diff
@@ -570,3 +607,31 @@ func (a *adapter) validateProcessingResources() *controller.ValidationResult {
 	}
 	return &controller.ValidationResult{Valid: true}
 }
+
+// validatePipelineRef checks that the release PipelineRun ref passes the checks from the ReleaseServiceConfig.
+func (a *adapter) validatePipelineRef() *controller.ValidationResult {
+	releasePlanAdmission, err := a.loader.GetActiveReleasePlanAdmissionFromRelease(a.ctx, a.client, a.release)
+	if err != nil {
+		if errors.IsNotFound(err) {
+			a.release.MarkValidationFailed(err.Error())
+			return &controller.ValidationResult{Valid: false}
+		}
+		return &controller.ValidationResult{Err: err}
+	}
+
+	releaseStrategy, err := a.loader.GetReleaseStrategy(a.ctx, a.client, releasePlanAdmission)
```

Comment:
I was about to suggest to create a `GetReleaseStrategyFromRelease`, but this resource is meant to disappear anyway.

Reply:
Yeah, I almost did create just that